test/integration/kafka_sarama_binding/kafka_test.go

```diff
@@ -101,6 +102,10 @@ func testSenderReceiver(t testing.TB) (func(), bindings.Sender, bindings.Receive
 		require.NoError(t, p.OpenInbound(context.TODO()))
 	}()
 
+	// Not perfect but we need to give OpenInbound() as chance to start
+	// as it's a race condition. I couldn't find something on 'p' to wait for
+	time.Sleep(5 * time.Second)
```

Comment:
IMHO this is fine for now. 

If I understand the race [condition](https://github.com/cloudevents/sdk-go/actions/runs/6589036256/job/17902601655#step:5:310) correctly, it is because we're using the `sarama` library in a way it might not be intended: there's a lot of concurrent access going on in this library and we are sharing the underlying client in a way during [setup](https://github.com/cloudevents/sdk-go/actions/runs/6589036256/job/17902601655#step:5:310) which leads to races after setting up the producer/consumer when calling `OpenInbound`.

Reply:
I can't speak to whether we're using Sarama correctly or not, but it seems golang was smart enough to notice that our tests might be trying to use `p` before it's actually ready. If I'm following things correctly, it does seem odd that call to `p.OpenInbound()` is blocking (thus the need for the thread), but there's no way to check to see when it's actually ready to be used. Feels like something is missing from Sarama to me.